10_rest/willshen8/cmd/puppy-server/main.go

```
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/willshen8/pkg/puppy"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "Puppy Json File Name").Short('d').ExistingFile()
+	port     = kingpin.Flag("port", "Port of the server").Short('p').Default("8888").Int()
+	storer   = kingpin.Flag("store", "Map/Sync Store").Short('s').Default("map").String()
+)
+
+// Config contains arguments parsed from commandline flags.
+type Config struct {
+	file   string
+	port   int
+	storer string
+}
+
+func main() {
+	cmdConfigs := parseCmdArgs(args)
+	newStorer, _ := createStorer(cmdConfigs)
+	decodedPuppies, _ := parseFileFlag(cmdConfigs)
```

why do swallow errors here?
what if I've specified a file that's not even JSON? 
Same for createStorer.

Also, IMO it should called something `parsePuppyFile(filename string)` probably better:
`parseJSONPuppies(r io.Reader)` - it will help with testing too (you could just use strings.NewReader("") to mock the input file.).

---

Done.